### PR TITLE
Add dark mode logo

### PR DESCRIPTION
**Pull request by trbromley**
Dark mode:
image

Light mode:
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +39,8 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/logo-dark-mode/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px">
```
Comment (trbromley) [inline README.md:43]:
```suggestion
  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px">
```
To update when merge ready

Comment (co9olguy) [reply README.md:43]: :pray: :tada:

**Review by josh146 — APPROVED** [review]
Thanks Tom! Approved ✔️ 

I realized that PyPI probably doesn't understand the `#gh-dark-mode-only` label, so _both_ images will display on the PyPI page: https://pypi.org/project/PennyLane

Having said that, I think the number of people reading about PL is much larger on GitHub than on PyPI.

**Comment by trbromley** [comment]
> both images will display on the PyPI page

Good point :thinking: do you think if we used a relative path for the dark mode image, then it would work in GitHub but not be visible in PyPI? I'm not sure how PyPI would render an image it couldn't load.

File: README.md [hunk 2]
```diff
@@ -39,7 +39,8 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px">
```
Comment (trbromley) [inline README.md:43]:
```suggestion
  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'">

```

File: README.md [hunk 3]
```diff
@@ -39,7 +39,9 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'">
```
Comment (trbromley) [inline README.md:43]:
```suggestion
  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'"/>

```

File: README.md [hunk 4]
```diff
@@ -39,7 +39,10 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'"/>
```
Comment (trbromley) [inline README.md:43]:
```suggestion
  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
```

File: README.md [hunk 5]
```diff
@@ -39,7 +39,10 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
+
+
```
Comment (trbromley) [inline README.md:45]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -39,7 +39,8 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
```
Comment (trbromley) [inline README.md:43]:
```suggestion
  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/logo-dark-mode/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
```

File: README.md [hunk 7]
```diff
@@ -39,7 +39,8 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/logo-dark-mode/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
```
Comment (trbromley) [inline README.md:43]:
```suggestion
  <img src="./pennylane/logo-dark-mode/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>

```

File: README.md [hunk 8]
```diff
@@ -39,7 +39,9 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
```
Comment (trbromley) [inline README.md:42]:
```suggestion
  <img src="./doc/_static/header.png#gh-light-mode-only" width="700px">

```

File: README.md [hunk 9]
```diff
@@ -39,7 +39,9 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="./doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="./pennylane/logo-dark-mode/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
+
```
Comment (trbromley) [inline README.md:44]:
```suggestion
```

File: README.md [hunk 10]
```diff
@@ -39,7 +39,8 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="./doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="./pennylane/logo-dark-mode/doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
```
Comment (trbromley) [inline README.md:43]:
```suggestion
  <img src="./doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>

```

File: README.md [hunk 11]
```diff
@@ -39,7 +39,8 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="./doc/_static/header.png#gh-light-mode-only" width="700px">
```
Comment (trbromley) [inline README.md:42]:
```suggestion
  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
```

File: README.md [hunk 12]
```diff
@@ -39,7 +39,8 @@
 
 <p align="center">
   <strong>Train a quantum computer the same way as a neural network.</strong>
-  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png" width="700px">
+  <img src="https://raw.githubusercontent.com/PennyLaneAI/pennylane/master/doc/_static/header.png#gh-light-mode-only" width="700px">
+  <img src="./doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
```
Comment (trbromley) [inline README.md:43]:
```suggestion
    <!--
    Use a relative import for the dark mode image. When loading on PyPI, this
    will fail automatically and show nothing.
    -->
    <img src="./doc/_static/header-dark-mode.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
```

**Comment by trbromley** [comment]
@josh146 I think the issue may be fixed in the current approach :+1:
- Use a relative path for dark mode image, which should fail on PyPI but not GH
- When failing to load, instead of showing a broken image we will just see nothing

**Comment by josh146** [comment]
That's awesome, thanks @trbromley and @antalszava!

**Pull request merged**